### PR TITLE
Vector binarizer

### PR DESCRIPTION
**Pull request by digital10111**
@ml-platform

### CODE REVIEW

File: adapters-1.6/src/main/java/com/flipkart/fdp/ml/adapter/VectorBinarizerModelAdapter.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.flipkart.fdp.ml.adapter;
+
+import com.flipkart.fdp.ml.modelinfo.IfZeroVectorModelInfo;
```
Comment (akshay-sharma) [inline adapters-1.6/src/main/java/com/flipkart/fdp/ml/adapter/VectorBinarizerModelAdapter.java]: remove unused imports

File: adapters-1.6/src/test/java/com/flipkart/fdp/ml/adapter/VectorBinarizerBridgeTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+/**
+ * Created by karan.verma on 09/11/16.
+ */
+public class VectorBinarizerBridgeTest extends SparkTestBase{
+    @Test
+    public void testVectorBinarizerDense() {
+        // prepare data
+
+        int[] sparseArray1 = {54, 76, 81};
```
Comment (akshay-sharma) [inline adapters-1.6/src/test/java/com/flipkart/fdp/ml/adapter/VectorBinarizerBridgeTest.java]: These are not used in this function. Please remove

File: models-info/src/main/java/com/flipkart/fdp/ml/transformer/VectorBinarizerTranformer.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private final VectorBinarizerModelInfo modelInfo;
+
+    public VectorBinarizerTranformer(final VectorBinarizerModelInfo modelInfo) {
+        this.modelInfo = modelInfo;
+    }
+
+    @Override
+    public void transform(Map<String, Object> input) {
+        Object value = input.get(modelInfo.getInputKeys().iterator().next());
+        double[] inp = (value == null)? null: (double[])value;
```
Comment (akshay-sharma) [inline models-info/src/main/java/com/flipkart/fdp/ml/transformer/VectorBinarizerTranformer.java]: Not sure if this is any different from this:
double[] inp = (double[])value;
Please verify

File: models-info/src/main/java/com/flipkart/fdp/ml/transformer/VectorBinarizerTranformer.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        input.put(modelInfo.getOutputKeys().iterator().next(), predict(inp, modelInfo.getThreshold()));
+    }
+
+    private double[] predict(double[] inp, Double threshold) {
```
Comment (akshay-sharma) [inline models-info/src/main/java/com/flipkart/fdp/ml/transformer/VectorBinarizerTranformer.java]: lets use "double threshold" instead of wrapper class "Double threshold".

**Comment by akshay-sharma** [comment]
remove .DS_Store file from commit.

**Comment by roshan-nair** [comment]
Add more tests to cover positive and negative threshold.

File: adapters-1.6/src/test/java/com/flipkart/fdp/ml/adapter/VectorBinarizerBridgeTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+/**
+ * Created by karan.verma on 09/11/16.
+ */
+public class VectorBinarizerBridgeTest extends SparkTestBase{
+
+    @Test(expected=IllegalArgumentException.class)
+    public void testVectorBinarizerNegativeThresholdValue() {
+        // prepare data
+
+        JavaRDD<Row> jrdd = sc.parallelize(Arrays.asList(
```
Comment (roshan-nair) [inline adapters-1.6/src/test/java/com/flipkart/fdp/ml/adapter/VectorBinarizerBridgeTest.java]: Just for clarity, can you remove the bits where you create the RDD, since the test doesn't require it here.

I think it will make the code easier to read.
